commands/utils/params.go

```diff
@@ -2,174 +2,180 @@ package utils
 	"strings"
+)
 
-	"github.com/jfrog/froggit-go/vcsclient"
-	"github.com/jfrog/froggit-go/vcsutils"
-	coreconfig "github.com/jfrog/jfrog-cli-core/v2/utils/config"
+const (
+	FrogbotConfigFile = "frogbot-config.yml"
 )
 
-type FrogbotParams struct {
-	JFrogEnvParams
-	GitParams
-	ScanPullRequestParams
-	WorkingDirectory   string
-	InstallCommandName string
-	InstallCommandArgs []string
-	RequirementsFile   string
+var configRelativePath = filepath.Join(".", ".jfrog", FrogbotConfigFile)
+
+type FrogbotConfigAggregator []FrogbotRepoConfig
+
+type FrogbotRepoConfig struct {
+	Server           coreconfig.ServerDetails
+	SimplifiedOutput bool
+	Params           `yaml:"params,omitempty"`
+}
+
+type Params struct {
+	Scan          `yaml:"scan,omitempty"`
+	Git           `yaml:"git,omitempty"`
+	JFrogPlatform `yaml:"jfrogPlatform,omitempty"`
 }
 
-type ScanPullRequestParams struct {
-	IncludeAllVulnerabilities bool
-	SimplifiedOutput          bool
-	FailOnSecurityIssues      bool
-	UseWrapper                bool
+type Project struct {
+	InstallCommandName  string   `yaml:"installCommandName,omitempty"`
+	InstallCommandArgs  []string `yaml:"installCommandArgs,omitempty"`
+	PipRequirementsFile string   `yaml:"pipRequirementsFile,omitempty"`
+	WorkingDirs         []string `yaml:"workingDirs,omitempty"`
+	UseWrapper          bool     `yaml:"useWrapper,omitempty"`
 }
 
-type JFrogEnvParams struct {
-	Server  coreconfig.ServerDetails
-	Project string
-	Watches string
+type Scan struct {
+	IncludeAllVulnerabilities bool      `yaml:"includeAllVulnerabilities,omitempty"`
+	FailOnSecurityIssues      *bool     `yaml:"failOnSecurityIssues,omitempty"`
+	Projects                  []Project `yaml:"projects,omitempty"`
 }
 
-type GitParams struct {
+type JFrogPlatform struct {
+	Watches         []string `yaml:"watches,omitempty"`
+	JFrogProjectKey string   `yaml:"jfrogProjectKey,omitempty"`
+}
+
+type Git struct {
 	GitProvider   vcsutils.VcsProvider
+	RepoName      string   `yaml:"repoName,omitempty"`
+	Branches      []string `yaml:"branches,omitempty"`
+	GitProject    string
 	RepoOwner     string
 	Token         string
-	Repo          string
-	BaseBranch    string
 	ApiEndpoint   string
 	PullRequestID int
 }
 
-func GetParamsAndClient() (*FrogbotParams, vcsclient.VcsClient, error) {
-	params, err := extractParamsFromEnv()
+func GetParamsAndClient() (configAggregator FrogbotConfigAggregator, server *coreconfig.ServerDetails, client vcsclient.VcsClient, err error) {
+	server, gitParams, err := extractEnvParams()
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
-	client, err := vcsclient.NewClientBuilder(params.GitProvider).ApiEndpoint(params.ApiEndpoint).Token(params.Token).Build()
-	return &params, client, err
-}
-
-func extractParamsFromEnv() (FrogbotParams, error) {
-	params := &FrogbotParams{}
-	extractGeneralParamsFromEnv(params)
+	defer func() {
+		e := SanitizeEnv()
+		if err == nil {
+			err = e
+		} else {
+			clientLog.Error(e)
```

Comment:
What is the reason we're adding this log messages here?
We need to make sure that every error message returned in any level of the flow gets logged automatically.
The same goes for the `updatePackageToFixedVersion` function.

Reply:
done